### PR TITLE
Improve reverse-proxy document and fix nginx config bug

### LINKED ISSUE #23711: [Doc] Gitea in sub-path config document is incompatible with path escaping
### Description

* Correct: https://docs.gitea.io/en-us/reverse-proxies/#nginx

* Buggy: https://docs.gitea.io/en-us/reverse-proxies/#nginx-with-a-sub-path

The ending slash shouldn't be used, because it causes nginx to unescape the path encodings, it breaks `%2f` in the path, breaks some Gitea git refs/paths and some Gitea package names/paths.

More details:

* https://github.com/go-gitea/gitea/issues/20890#issuecomment-1483950304

Thanks to @ghnp5

### LINKED ISSUE #24612: Docs: Reverse Proxy Documentation should mention client_max_body_size for nginx
### Description

I had problems with Gitea behind an nginx reverse proxy when trying to push big git LFS files.
I configured nginx like described in https://docs.gitea.io/en-us/administration/reverse-proxies/#nginx

The problem apparently was that nginx has a default size limit for uploading via http (at least in the default configuration on Ubuntu?), which can be overriden with `client_max_body_size` (e.g. `client_max_body_size 0; for "no limit").
Example:
```
server {
    listen 80;
    server_name git.example.com;
    # DG: disable size checking for uploads to prevent LFS problems
    client_max_body_size 0;

    location / {
        proxy_pass http://127.0.0.1:3000;
        ...
```

### Gitea Version

any

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

behind a nginx reverse proxy

### Database

None

### PR DESCRIPTION
**Pull request by wxiaoguang**
Close #23711, thanks to @ghnp5 !
Close  #24612, thanks to @DanielGibson !

Major changes:

* the default value of nginx's client_max_body_size is too small, so put a 512M here
* move `Resolving Error: 413 Request Entity Too Large` to a sub-section of `Nginx` section
* make nginx use unescaped the URI and keep "%2F" as is when using sub-path
* add details for General sub-path configuration

### CODE REVIEW

**Review by yardenshoham — APPROVED** [review]

File: docs/content/doc/administration/reverse-proxies.en-us.md [hunk 1]
```diff
@@ -40,23 +41,31 @@ server {
...
-        # Note: Trailing slash
-        proxy_pass http://localhost:3000/;
-        proxy_set_header Host $host;
-        proxy_set_header X-Real-IP $remote_addr;
-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-        proxy_set_header X-Forwarded-Proto $scheme;
+    location /gitea/ {
+        client_max_body_size 512M;
+
+        # make nginx use unescaped the URI, keep "%2F" as is
```
Comment (wxiaoguang) [inline docs/content/doc/administration/reverse-proxies.en-us.md]:
```suggestion
        # make nginx use unescaped URI, keep "%2F" as is
```

**Comment by silverwind** [comment]
Am a bit sceptical regarding thowse rewrites, but I don't know another solution off my head at least because nginx slash handling is generally weird.

**Comment by DanielGibson** [comment]
I think the documentation on `client_max_body_size` could be more explicit about it not only affecting file uploads in the webui, but also via git when using LFS (=> keep the `If you use Git LFS, this will also limit the size of the largest file you will be able to push.` sentence from the prior version).
For lazy people who don't want to look at the nginx doc you could also mention the special case of `0` meaning no limit ;)

**Review by lunny — APPROVED** [review]

File: docs/content/doc/administration/reverse-proxies.en-us.md [hunk 2]
```diff
@@ -40,23 +41,31 @@ server {
 }
 ```
 
+### Resolving Error: 413 Request Entity Too Large
+
+This error indicates nginx is configured to restrict the file upload size,
+you can fine tune the `client_max_body_size` option according to [nginx document](http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size).
```
Comment (wxiaoguang) [inline docs/content/doc/administration/reverse-proxies.en-us.md]:
```suggestion
This error indicates nginx is configured to restrict the file upload size,
it affects attachment uploading, form posting, package uploading and LFS pushing, etc.
You can fine tune the `client_max_body_size` option according to [nginx document](http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size).
```

**Comment by wxiaoguang** [comment]
> For lazy people who don't want to look at the nginx doc you could also mention the special case of `0` meaning no limit ;)

8298848fe3514002bfd8e58d880793ec58e9d27c


> I think the documentation on `client_max_body_size` could be more explicit about it not only affecting file uploads in the webui,

"nolimit" would fill up user's disk if the user runs nginx with  tmp dir  in a small file system.

**Review by wolfogre — APPROVED** [review]

File: docs/content/doc/administration/reverse-proxies.en-us.md [hunk 3]
```diff
@@ -387,3 +389,13 @@ gitea:
...
+
+## General sub-path configuration
+
+Usually it's not recommended to put Gitea in a sub-path, it's not widely used and may have some issues in rare cases.
+
+If you really need to do so, to make Gitea works with sub-path (eg: `http://example.com/gitea/`), here are the requirements:
+
+1. Set `[server] ROOT_URL = http://example.com/gitea/` in your `app.ini` file.
+2. Make the reverse-proxy pass `http://example.com/gitea/foo` to `http://gitea-server:3000/foo`.
+3. Make sure the reverse-proxy do not escape the URI, the request `http://example.com/gitea/a%2Fb` should be passed as `http://gitea-server:3000/a%2Fb`.
```
Comment (wxiaoguang) [inline docs/content/doc/administration/reverse-proxies.en-us.md]:
```suggestion
3. Make sure the reverse-proxy not decode the URI, the request `http://example.com/gitea/a%2Fb` should be passed as `http://gitea-server:3000/a%2Fb`.
```

**Pull request merged**